### PR TITLE
fix(store): refuse to remove a store that contains another store

### LINKED ISSUE #1879: `store remove` deletes a different registered store nested inside the target
## Description

`openspec store remove <id>` deletes the store's folder with a recursive
`fs.rm`. Before deleting, it checks only the target's own metadata and id. It
never checks whether another *registered* store lives inside that folder.

`openspec store register` accepts exactly that layout. The natural way to get
there is a shared store vendored into a team store as a git submodule. Removing
the outer store then:

- deletes the inner store, including planning work that was never committed
- exits 0 with an empty `status`, so nothing says a second store went with it
- leaves the inner store registered, pointing at a path that no longer exists

`store setup` refuses a nested repository (`store_setup_inside_git_repo`), but
`store register` does not. `store remove` turns that disagreement into data
loss.

## Reproduction

```bash
openspec store setup plat --path ~/upstream/plat --json
openspec store unregister plat --json
openspec store setup team-plans --path ~/openspec/team-plans --json

# Vendor the shared store into the team store, then register the vendored copy
git -C ~/openspec/team-plans -c protocol.file.allow=always \
  submodule add ~/upstream/plat vendor/plat
git -C ~/openspec/team-plans commit -m "vendor plat"
openspec store register ~/openspec/team-plans/vendor/plat --json

# Some planning work that exists only on disk
mkdir -p ~/openspec/team-plans/vendor/plat/openspec/changes/draft-idea
echo "# Draft (uncommitted work)" \
  > ~/openspec/team-plans/vendor/plat/openspec/changes/draft-idea/proposal.md

openspec store remove team-plans --yes --json
```

## Expected

`remove team-plans` refuses, and names the store inside it, or at the very least
leaves `plat` alone. `docs/cli.md` scopes `remove` to one store: "Forget a local
store registration and delete its local folder."

## Actual

```
[nested]  register plat at .../openspec/team-plans/vendor/plat rc=0
    remove team-plans rc=0 :: "deleted_path":".../openspec/team-plans" "status":[]
    plat checkout still on disk? NO   draft proposal? NO
    registry still lists plat? ['plat']
```

Control, with `plat` cloned beside `team-plans` instead of inside it:

```
[sibling] register plat at .../work/plat rc=0
    remove team-plans rc=0 :: "deleted_path":".../openspec/team-plans" "status":[]
    plat checkout still on disk? yes   draft proposal? yes
```

## Root cause

- `assertSafeToDeleteStoreRoot` (`src/core/store/operations.ts:905`) checks the
  target's metadata and id only.
- `removeStore` then runs `fs.rm(removed.storeRoot, { recursive: true, force:
  true })` (`src/core/store/operations.ts:983`).
- `registerExistingStore` (`src/core/store/operations.ts:737`) and
  `registerStore` (`src/core/store/registry.ts:323`) accept a checkout inside
  another store's folder.

## Impact

Irreversible loss of a store the user did not name, including uncommitted
planning work, and a registry entry that now points at nothing. The command
reports success.

## Environment

- OpenSpec 1.13.0 (reproduced on the published npm package and on a build of
  `main` @ `9d4e597`)
- Node 22, Linux
- Every run used its own temporary `HOME`, `XDG_CONFIG_HOME` and `XDG_DATA_HOME`

### PR DESCRIPTION
**Pull request by dwin-gharibi**
Closes #1879.

## Why

`openspec store remove <id>` deletes the store folder with
`fs.rm(..., { recursive: true, force: true })`. Its safety check,
`assertSafeToDeleteStoreRoot`, only confirms that the folder carries the
target's own metadata. It never asked whether another *registered* store lives
inside that folder.

Supported commands reach that layout. `store register` accepts a checkout inside
another store, and the natural case is a shared store vendored into a team store
as a git submodule. Removing the outer store then:

- deleted the inner store, including planning work that was never committed
- exited 0 with `"status": []`, so nothing said a second store went with it
- left the inner store registered at a path that no longer exists

Verified with stores created by `openspec store setup` and `store register`: the
vendored store's draft proposal was gone after `store remove team-plans --yes`,
and `store list` still showed it.

## What Changes

- `removeStore` refuses with a new error code,
  `store_remove_contains_registered_store`, when any other registration's root
  is inside the folder about to be deleted.
  - The message names each nested store's id and path.
  - The fix says to run `openspec store unregister <nested-id>` first, or
    `openspec store unregister <id>` to forget the target without deleting
    files.
- The check runs in the existing `beforeCommit` hook, under the registry lock
  that commits the removal, next to the metadata check that already lives there.
  `unregisterStoreRegistration` now passes the remaining registrations to that
  hook.
- Paths are compared canonically. A registration recorded through a symlinked
  path is still found, and a sibling that merely shares a name prefix
  (`team-plans-archive`) is not mistaken for a nested store.
- A stale nested registration, whose folder is already gone, also blocks the
  removal. Otherwise the registry would end up pointing into a deleted folder.
- `docs-lab/reference/cli.md` and the error-code list in `docs/agent-contract.md` document the
  refusal.

Unchanged: removing a store whose other registrations are siblings,
`store unregister`, and `store register`.

### Why not refuse nesting in `store register` too?

A vendored store is a legitimate layout. Registering the submodule is how a team
works in the shared store. Refusing it at register time would break that
workflow to prevent a hazard that only exists at delete time. The data loss comes
from `remove`, so the guard goes there.

## Testing

`test/commands/store-remove-nested.test.ts`: 10 tests, run through the built
CLI. Written first and watched fail:

- **Before the fix (tests only, on `9d4e597`):** 7 failed, 3 passed. Every
  refusal test failed because `remove` exited 0 and deleted the nested store
  (`expected +0 to be 1`, and `store` non-null in the JSON). The three tests
  that do not depend on the bug passed: the sibling control, the name-prefix
  sibling, and remove after unregister.
- **After the fix:** 10 passed.

### CI parity

Run in an isolated sandbox on `9d4e597` under Node 20.19.0 and pnpm 10.34.5,
with the steps from `.github/workflows/ci.yml`:

| Step | Result |
|---|---|
| `pnpm run build` | pass |
| `pnpm exec tsc --noEmit` | pass |
| `pnpm lint` | pass |
| `pnpm test` (`VITEST_MAX_WORKERS=4`) | 4564 passed, 9 failed (4573) |
| `pnpm test` on clean `9d4e597`, same sandbox | 4555 passed, 8 failed (4563) |

The full-suite failures are all timeouts — `Test timed out in 10000ms`, or
`spawnSync npm ETIMEDOUT` in the npm packing test — in CLI-heavy files this
change does not touch: `store-references`, `store-remote`,
`store-root-selection`, `workset`, `package-install-scripts` and
`store.test.ts`. Two checks say the sandbox causes them, not this change:

- clean `9d4e597` fails the same files the same way on the same machine
- every one of those files passes alone (`--maxWorkers=2`), on this branch and
  on clean `9d4e597`: `store-references` 8/8, `store-remote` 19/19,
  `store-root-selection` 44/44, `store.test.ts` 43/43, `workset` 41/41,
  `package-install-scripts` 8/8

The branch adds exactly its 10 new tests: 4573 = 4563 + 10.

Edge cases covered:

- control: another registered store beside the target is left alone, and the
  target is removed
- nested store: refused, and the target folder, the nested draft and both
  registrations are all intact
- the JSON refusal's shape: `store`/`files` null, one `error` with the new code,
  the nested id and path in the message, and the `unregister` command in the fix
- human mode prints the refusal and the fix on stderr, and deletes nothing
- several nested stores are all named
- a sibling that shares the name prefix is not treated as nested
- a stale nested registration still blocks the removal
- after `store unregister <nested>`, removing the outer store works
- a nested registration recorded through a symlinked path is found. This test
  is skipped on Windows, where creating a directory symlink needs elevated
  rights.
- a real git submodule with an uncommitted draft, the reported scenario

## Changeset

`.changeset/store-remove-keeps-nested-stores.md` (patch).

## Notes for review

`fix-store-setup-no-init-git` also edits `src/core/store/operations.ts`, but in
`prepareSetupPlan`, far from `removeStore`. The two should merge without
conflicts.

## Summary by CodeRabbit

- **Bug Fixes**
  - `openspec store remove` now refuses to delete a store containing another registered store, protecting nested stores and uncommitted work.
  - Error messages identify the nested store and provide the `openspec store unregister` command needed to proceed.
  - Nested stores are detected through symlinks, Git submodules, and stale registrations, while sibling stores remain unaffected.

- **Documentation**
  - Updated CLI and diagnostic documentation to describe the new removal safeguards and resolution steps.

### CODE REVIEW

File: src/core/store/operations.ts [hunk 1]
```diff
@@ -963,9 +997,12 @@ export async function removeStore(
     id,
     expectedBackend: target.backend,
     globalDataDir: target.globalDataDir,
-    beforeCommit: async (entry) => {
+    beforeCommit: async (entry, remaining) => {
       const safeTarget = await assertSafeToDeleteStoreRoot(entry.storeRoot, id);
       rootMissing = !safeTarget.exists;
+      if (safeTarget.exists) {
+        assertNoRegisteredStoreInside(entry.storeRoot, id, remaining);
```
[the remark this answers is not in the document]
Comment (clay-good) [reply src/core/store/operations.ts:1004]: Out of scope for this PR. The gap between the registry commit and `fs.rm` is pre-existing: on main, `removeStore` already releases the registry lock before the recursive delete, so a registration committed inside the target during that window is deleted the same way. This PR does not widen that window; it closes the non-racy case (a nested store that was already registered, which main deletes every time). A root-scoped removal reservation that `register` honors would touch the registration path and the lock design, so it belongs in a separate change.

**Comment by clay-good** [comment]
Hardening pass, one commit on top of the fix:

- 44393b7b: moved the nested-store refusal note from legacy `docs/cli.md` to the canonical `docs-lab/reference/cli.md` (store remove section). `docs/agent-contract.md` keeps its error-code entry. docs-lab changes need @TabishB review.

Verified: the issue repro on main exits 0, deletes the nested store's draft, and leaves `plat` registered. On this branch it exits 1 with `store_remove_contains_registered_store` and keeps both. The test file fails 7/10 with `operations.ts`/`registry.ts` reverted and passes 10/10 with them. `store.test.ts` passes; tsc and lint are clean.

CodeRabbit's remove/register race is pre-existing on main (the lock is released before `fs.rm` there too), so I replied and resolved it as out of scope. Merge order with #1884: both touch `src/core/store/operations.ts` and `docs-lab/reference/cli.md` in different sections, and a trial three-way merge of the two branches is conflict-free in either order.

**Review by alfred-openspec — APPROVED** [review]
Removal now checks the remaining registrations under the registry lock, compares canonical paths, and refuses before deleting nested stores while preserving sibling and unregister behavior. Focused tests pass. @TabishB remains requested for final docs review.
